modules/transport-netty4/src/test/java/org/elasticsearch/rest/Netty4BadRequestIT.java

```diff
@@ -101,4 +101,27 @@ public void testInvalidHeaderValue() throws IOException {
         assertThat(map.get("type"), equalTo("content_type_header_exception"));
         assertThat(map.get("reason"), equalTo("java.lang.IllegalArgumentException: invalid Content-Type header []"));
     }
+
+    public void testInvalidHeaderCombinations() throws IOException {
+        final Request request = new Request("GET", "/_cluster/settings");
+        final RequestOptions.Builder options = request.getOptions().toBuilder();
+        options.addHeader("Content-Type", "application/vnd.elasticsearch+json;compatible-with=7");
+        options.addHeader("Accept", "application/vnd.elasticsearch+json;compatible-with=8");
+        request.setOptions(options);
+        request.setJsonEntity("{\"transient\":{\"search.*\":null}}");
+
+        final ResponseException e = expectThrows(ResponseException.class, () -> client().performRequest(request));
+        final Response response = e.getResponse();
+        assertThat(response.getStatusLine().getStatusCode(), equalTo(400));
+        final ObjectPath objectPath = ObjectPath.createFromResponse(response);
+        final Map<String, Object> map = objectPath.evaluate("error");
+        assertThat(map.get("type"), equalTo("compatible_api_headers_combination_exception"));
+        assertThat(map.get("reason"), equalTo("Content-Type and Accept headers have to match when content is present. " +
```

Comment:
how much details do we want on this exception?
at the moment it helped debugging tests when seeing which failed validation

Reply:
I think this is a good level of detail.